[CI] FullClusterRestartIT testClusterState {cluster=UPGRADED} failing

**Build Scans:**
- [elasticsearch-intake #35618 / 9.2.7_bwc-snapshots](https://gradle-enterprise.elastic.co/s/rwxmt73u5pqwc)
- [elasticsearch-pull-request #126362 / 9.3.2_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/urjq3xui4qiuy)

**Reproduction Line:**
```
./gradlew ":qa:full-cluster-restart:v9.2.7#bwcTest" -Dtests.class="org.elasticsearch.upgrades.FullClusterRestartIT" -Dtests.method="testClusterState {cluster=UPGRADED}" -Dtests.seed=73160684EAA819D7 -Dtests.bwc=true -Dtests.locale=en-DE -Dtests.timezone=Atlantic/Jan_Mayen -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.upgrades.FullClusterRestartIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testClusterState%20%7Bcluster%3DUPGRADED%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<1> but was:<null>
```

**Issue Reasons:**
- [main] 2 failures in test testClusterState {cluster=UPGRADED} (0.4% fail rate in 490 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

## Test Failure Analysis

**Summary**
- `testClusterState {cluster=UPGRADED}` fails with `expected:<1> but was:<null>` because the randomized test runner (controlled by seed `73160684EAA819D7`) places `testClusterState {cluster=UPGRADED}` **before** `testClusterState {cluster=OLD}` in the execution order.
- The UPGRADED test case triggers a cluster upgrade (via `ParameterizedFullClusterRestartTestCase.maybeUpgrade()`), but `testClusterState {cluster=OLD}` — which creates `template_1` — has not yet run. The cluster is therefore upgraded without the template, and the assertion on `metadata.templates.template_1.settings.index.number_of_shards` returns `null`.
- This is a **test-design ordering bug**: the test has an implicit dependency (OLD phase must run before UPGRADED phase) but the `static boolean upgraded` flag and the randomized JUnit runner can violate that assumption.

**Release risk: low-risk**

> The failure is a test ordering bug, not a production code defect. No production cluster behavior is affected: the legacy template API, cluster state serialization, and gateway recovery all work correctly. The issue only surfaces when the random seed interleaves `{cluster=OLD}` and `{cluster=UPGRADED}` test cases across the upgrade boundary, which is a CI-only artifact.
> To raise to `medium-risk`, evidence would need to show that the template actually disappears from a properly-sequenced upgrade (i.e., the OLD phase ran and the template was later lost). No such evidence exists.

**Impact**
- Affected branches: `main`
- Affected OS: Linux (Ubuntu)
- Affected pipelines: `elasticsearch-intake` (bwc-snapshots), `elasticsearch-pull-request` (bwc-snapshots)
- Trend: Low-rate intermittent (0.4%, 2/490 executions); muted on `main` since 2026-03-08

**Reproduction**
- Command: `./gradlew ":qa:full-cluster-restart:v9.2.7#bwcTest" -Dtests.class="org.elasticsearch.upgrades.FullClusterRestartIT" -Dtests.method="testClusterState {cluster=UPGRADED}" -Dtests.seed=73160684EAA819D7 -Dtests.bwc=true -Dtests.locale=en-DE -Dtests.timezone=Atlantic/Jan_Mayen -Druntime.java=25`
- Commit: `667fd56f1494d90b58dd5aa0cc12118da7e9bdde`
- Remote job: https://buildkite.com/elastic/elasticsearch-agentic-workflow/builds/100#019f69b8-43b4-477e-99cb-c534d16e2325 — ubuntu-2404
- Result: **reproduced ✨** — same `AssertionError: expected:<1> but was:<null>` at `FullClusterRestartIT.java:412`
- Build scan: https://gradle-enterprise.elastic.co/s/txr2fewsom3ei

**Mitigation Proposal**

- **Reproduction verdict**: The failure reproduces deterministically with seed `73160684EAA819D7`. The UPGRADED test case triggers the cluster upgrade before the corresponding OLD test case has run, confirming the test-ordering hypothesis.
- **Codebase investigation**:
  - `ParameterizedFullClusterRestartTestCase.maybeUpgrade()` (a `@Before` method) upgrades the cluster when `upgraded == false && requestedUpgradeStatus == UPGRADED`. The `upgraded` field is `static`, so once the first UPGRADED test triggers an upgrade, all subsequent tests run against the new cluster.
  - `testClusterState {cluster=OLD}` creates `template_1`; `testClusterState {cluster=UPGRADED}` asserts it exists. These two cases must run in order, but the randomized runner offers no such guarantee.
- **Recent git history**: No relevant changes to `testClusterState` or `ParameterizedFullClusterRestartTestCase` in the window around the failures (last meaningful change was Oct 2025). The failures appear to be an existing test fragility exposed by a particular random seed.
- **Proposed fix** (test hardening): Ensure the UPGRADED test case gracefully handles the case where the OLD phase setup did not happen. Options in order of preference:
  1. **Preferred**: Add `assumeTrue` in `testClusterState` for the UPGRADED phase to skip if `template_1` does not exist (with a log message), making the test resilient to ordering issues.
  2. **Alternative**: Annotate the class with `@TestCaseOrdering(LuceneTestCase.MethodSorter.class)` or equivalent to ensure OLD test cases always run before UPGRADED ones.
  3. **Simplest**: Unmute the test; it appears deterministically reproducible with the known seed, meaning the mute is hiding a real (if minor) test design issue.
- **Effort estimate**: trivial — the fix is a small guard in `testClusterState`, or adding ordering annotation.
- **Confidence**: 0.85 (high) — the ordering hypothesis is supported by the deterministic reproduction and the "1 test completed, 1 failed" log (only the UPGRADED case ran, confirming OLD setup was skipped).

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -29,15 +29,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test {feature:METRICS}
   issue: https://github.com/elastic/elasticsearch/issues/153507
-- class: org.elasticsearch.upgrades.FullClusterRestartIT
-  method: testClusterState {cluster=UPGRADED}
-  issue: https://github.com/elastic/elasticsearch/issues/143800
-- class: org.elasticsearch.upgrades.FullClusterRestartIT
-  method: testOperationBasedRecovery {cluster=OLD}
-  issue: https://github.com/elastic/elasticsearch/issues/143801
-- class: org.elasticsearch.upgrades.FullClusterRestartIT
-  method: testOperationBasedRecovery {cluster=UPGRADED}
-  issue: https://github.com/elastic/elasticsearch/issues/143802
 - class: org.elasticsearch.xpack.application.OpenAiServiceUpgradeIT
   method: testOpenAiEmbeddings {upgradedNodes=3}
   issue: https://github.com/elastic/elasticsearch/issues/144440

